hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -532,6 +532,13 @@ public boolean isBeforeTimelineStarts(String instant) {
         && compareTimestamps(instant, LESSER_THAN, firstNonSavepointCommit.get().getTimestamp());
   }
 
+  @Override
+  public boolean isBeforeTimelineStartsByCompletionTime(String completionTime) {
+    Option<HoodieInstant> firstNonSavepointCommit = getFirstNonSavepointCommit();
```

Comment:
`getFirstNonSavepointCommit()`uses the sorted instants by (start) instant time.  Here it needs the instant with the smallest completion time.

Reply:
Good catch! Will fix this